src/users/cd-login.vue

```diff
@@ -1,10 +1,29 @@
 <template>
-  <div class="cd-login">
-    <form @submit.prevent="login">
-      <input type="email" v-model="email" />
-      <input type="password" v-model="password" />
-      <input type="submit" value="Login" />
-    </form>
+  <div class="cd-login container-fluid">
+    <div class="row">
+      <h3 class="cd-login__header text-center">{{ $t('Login') }}</h3>
+    </div>
+    <div class="row">
+      <div class="cd-login__box">
+        <form @submit.prevent="login">
+          <div class="form-group">
+            <label>{{ $t('Email') }}</label>
+            <input class="form-control" data-vv-name="email" data-vv-validate-on="blur" type="email" v-model="email" v-validate="'required|email'" novalidate/>
+          </div>
+          <p class="cd-login__email-req-err text-danger" v-show="errors.has('email:required')">{{ $t('Email is required') }}</p>
+          <p class="cd-login__email-format-err text-danger" v-show="errors.has('email:email')">{{ $t('Email should be in the format: janedoe@example.com') }}</p>      
+          <div class="form-group">
+            <label>Password</label>
+            <input class="form-control" data-vv-name="password" data-vv-validate-on="blur" type="password" v-model="password" v-validate="'required'" />
+          </div>
+           <p class="cd-login__password-req-err text-danger" v-show="errors.has('password:required')">{{ $t('Password is required') }}</p>
+          <input class="cd-login__button btn btn-primary" type="submit" value="Login" />
+        </form>
+        <p v-show="errors.has('loginFailed')" class="cd-login__login-failed text-danger">{{ $t('There was a problem logging in: Invalid email or password') }}</p>
```

Comment:
Invalid email or password is an existing string, i'd prefer we use it as a variable so that we can keep the translation live and save the translators some work

Reply:
cc @DanielBrierton

Reply:
no, more like this -> https://github.com/CoderDojo/cp-zen-frontend/blob/master/src/dojos/cd-dojo-details.vue#L42

Reply:
I have this {{ $t('There was a problem logging in: {msg}', {msg: 'Invalid email or password!'}) }}. Am I meant to have it as a variable with a different syntax?